classes/subject.py

```diff
@@ -1,12 +1,26 @@
-from enum import Enum
-        
-class SubjectType(Enum):
-    mother = 1
-    child = 2
-    alledged_father = 3
+import re
+
+from classes.genotype import Genotype
+
+# from enum import Enum
+# class SubjectType(Enum):
+#      mother = 1
+#      child = 2
+#      alledged_father = 3
 
 class Subject():
-    def __init__(self, type: SubjectType, name: str, genetic_profile: list[str]):
-        self.type = type
+    type: str
+    
+    def __init__(self, name: str, genotype: list[Genotype]):
         self.name = name
-        self.genetic_profile = genetic_profile
```

Comment:
Genotype can be referred to either a singular locus+alleles, or a handful of locus+alleles all together.
It is not wrong to rename the `genetic_profile` to `genotype`, but it cause some confusion. I believe it's better to revert this change.